kaydle-primitives/src/node.rs

```diff
@@ -497,7 +497,12 @@ impl<'i> NodeList<'i> for Children<'i, '_> {
             )
         }
 
-        self.run_parser(parse_node_start(char('}').value(())))
+        let child_terminator = char('}')
```

Comment:
I believe that, because the child-terminator *must* be followed by *any* node terminator, it'd be more appropriate to use `parse_node_terminator` non-optionally here:

```rust
// The end-of-child must also include the end-of-node for the node
// it belongs to
let child_terminator = char('}')
    .precedes(parse_node_space.opt())
    .precedes(parse_node_terminator)
```

I believe that both my code and your proposed change will incorrectly accept this document:

```
node {
    child1 {
        sub-child
    } child2
}
```